astropy/io/fits/tests/test_hdulist.py

```diff
@@ -913,3 +914,38 @@ def test_proper_error_raised_on_non_fits_file_with_unicode(self):
         # This should raise an IOError because there is no end card.
         with pytest.raises(IOError):
             fits.open(filename)
+
+    # Avoid a spurious pyflakes warning about an unused import
+    # of six by adding a reason string
+    @pytest.mark.skipif(six.PY2, reason='six.PY2')
+    @pytest.mark.parametrize('ignore', [False, True])
+    @pytest.mark.parametrize('fileobj', [False, True])
```

Comment:
Repsonse to @pllim comment

I originally had two tests, one where the file was passed as a filename and one where it was passed as a file object. To avoid some code repetition I combined them into a single test, and used a boolean flag (the `fileobj` parameter) to determine what was to be done. This was partly in response to my mis-understanding of a comment by @saimn